instrumentation/opentelemetry-instrumentation-fastapi/src/opentelemetry/instrumentation/fastapi/__init__.py

```diff
@@ -358,6 +359,10 @@ def uninstrument_app(app: fastapi.FastAPI):
         app.middleware_stack = app.build_middleware_stack()
         app._is_instrumented_by_opentelemetry = False
 
+        # Remove the app from the set of instrumented apps to prevent memory leaks
```

Comment:
Since WeakSet accomplishes avoiding memory leaks here, I think this is more `Avoid calling u instrument twice if the instrumentation is later disabled` or such